lib/elixir/src/elixir_def.erl

```diff
@@ -61,6 +61,7 @@ fetch_definitions(File, Module) ->
 
   {All, Private} = fetch_definition(Entries, File, Module, Set, Bag, [], []),
   Unreachable = elixir_locals:warn_unused_local(File, Module, All, Private),
```

Comment:
Minor nitpick: what do you think about reordering this to?

```erlang
{All, Private} = fetch_definition(Entries, File, Module, Set, Bag, [], []),
elixir_locals:ensure_no_undefined_local(File, Module, All),
elixir_locals:ensure_no_import_conflict(File, Module, All),
Unreachable = elixir_locals:warn_unused_local(File, Module, All, Private),
{All, Unreachable}.
```

So it fails faster in case of undefined functions and import conflicts, and the variable assignment is closer to its use?

Reply:
I would prefer to leave the errors for last, especially because the unused warning alongside the undefined local error may reveal things such as misplaced arity.

Reply:
Oh! I see. It makes sense.